packages/apollo-gateway/src/__tests__/gateway/executor.test.ts

```diff
@@ -25,30 +42,46 @@ describe('ApolloGateway executor', () => {
       queryHash: 'hashed',
       context: null,
       cache: {} as any,
+      logger,
     });
 
     expect(errors![0].message).toMatch(
-      'Variable "$first" got invalid value "3"; Expected type Int.',
+      'Variable "$first" got invalid value "3";',
     );
   });
 
   it('still sets the ApolloServer executor on load rejection', async () => {
-    jest.spyOn(console, 'error').mockImplementation();
-
     const gateway = new ApolloGateway({
-      // Empty service list will throw, which is what we want.
+      // Empty service list will trigger the gateway to crash on load, which is what we want.
       serviceList: [],
+      logger,
     });
 
+    // Mock implementation of process.exit with another () => never function.
+    // This is because the gateway doesn't just throw in this scenario, it crashes.
+    const mockExit = jest
+      .spyOn(process, 'exit')
+      .mockImplementation((code) => {
+        throw new Error(code?.toString());
+      });
+
     const server = new ApolloServer({
       gateway,
       subscriptions: false,
+      logger,
     });
 
     // Ensure the throw happens to maintain the correctness of this test.
     await expect(
       server.executeOperation({ query: '{ __typename }' })).rejects.toThrow();
 
     expect(server.requestOptions.executor).toBe(gateway.executor);
+
+    expect(logger.error.mock.calls).toEqual([
+      ["Error checking for changes to service definitions: Tried to load services from remote endpoints but none provided"],
+      ["This data graph is missing a valid configuration. 1"]
```

Comment:
I guess `1` is the `code` from `process.exit`'s callback above?  Maybe we could just have the thrown error in these tests be prefixed with something (e.g. `Exit code: ${code?.toString()}`) just to make this stray `1` in this error seem more self-explanatory?

Reply:
No longer applies / resolved by afa6746